examples/wallet/program/src/main.rs

```diff
@@ -56,32 +56,20 @@ impl WalletProgram for Wallet {
 }
 
 impl VerifiableTemplate for Wallet {
-  fn verify(&self, tx: &[u8], signature: &[u8; 64]) -> bool {
+  fn verify(&self, tx: alloc::vec::Vec<u8>, signature: [u8; 64]) -> bool {
     // Check that the transaction is signed by the owner
     let public_key = VerifyingKey::from_bytes(&self.owner.0).unwrap();
-    let signature = Signature::from_bytes(signature);
+    let signature = Signature::from_bytes(&signature);
     public_key.verify(&tx, &signature).is_ok()
   }
 }
 
-#[cfg(all(
-  any(target_arch = "riscv32", target_arch = "riscv64"),
-  target_feature = "e",
-  test
-))]
-mod test {
-  use super::*;
-
-  #[test]
-  fn test_wallet() {
-    let owner = Pubkey([0u8; 32]);
-    let wallet = Wallet::new(owner);
-
-    Wallet::spawn(owner);
-    let send_arguments = SpendArguments {
-      recipient: [0u8; 24],
-      amount: 0,
-    };
-    wallet.spend(send_arguments);
-  }
+#[link_section = ".text.athexp.verify"]
```

Comment:
Because it requires rethinking and probably many changes in the macro. The proc macro needs to be placed on `impl WalletProgram for Wallet` and it cannot be placed in the same file twice because it would cause all sorts of errors like registering a global allocator twice, etc. I don't want to mix it in this PR - it can be done later.